multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -87,16 +87,15 @@ func NewLabelIdentityReconciler(
 // +kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
 func (r *LabelIdentityReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	klog.V(2).InfoS("Reconciling Pod for label identity", "pod", req.NamespacedName)
-	var commonArea commonarea.RemoteCommonArea
-	var err error
-	commonArea, r.localClusterID, err = r.commonAreaGetter.GetRemoteCommonAreaAndLocalID()
-	if commonArea == nil {
-		return ctrl.Result{Requeue: true}, err
+	if r.remoteCommonArea == nil {
+		commonArea, localClusterID, err := r.commonAreaGetter.GetRemoteCommonAreaAndLocalID()
+		if commonArea == nil {
+			return ctrl.Result{Requeue: true}, err
+		}
+		r.remoteCommonArea, r.localClusterID = commonArea, localClusterID
```

Comment:
It can fix the data race in unit test but not the race in production as `Reconcile` can be executedly in parallel.

Reply:
That is true. Fixed. I also discovered that the same problem exists in `serviceexport_controller.go` as well. I'll let you and @luolanzone evaluate if we should backport this

Reply:
I feel we should backport this, but maybe submit the fix for `serviceexport_controller.go` in a new PR to make backport easier? @tnqn any suggestion?

Reply:
agreed

Reply:
@Dyanngg Could you create a separate PR for serviceexport? I think gateway_controller.go also has this issue, can you fix it in the new PR as well? let me know if you need me to create the PR. Thanks.

Reply:
I have created another PR #4305. `gateway_controller.go` does not have this issue now since the max allowed reconcile is 1. There is also a TODO for add locking if this is changed later.